runtime/compiler/optimizer/InterpreterEmulator.cpp

```diff
@@ -59,17 +275,25 @@ InterpreterEmulator::maintainStackForIf(TR_J9ByteCode bc)
       canFallThru = !canBranch;
       }
 
+   // The branch target can be successor of the fall through, so gen fall through block first such
+   // that the predecessor is interpreted before the successor in order to propagate the operand
+   // stack and local slots state.
+   // This doesn't work when the fall through contain control flow, but there is no functional issue
+   // as the object info won't be propagated if there exists unvisited predecessor. This will be
+   // fixed when we traverse the bytecodes in reverse post order at CFG level.
+   //
+   if (canFallThru)
+      {
+      debugTrace(tracer(), "maintainStackForIf canFallThrough to bcIndex=%d\n", fallThruBC);
+      genTarget(fallThruBC);
```

Comment:
@vijaysun-omr This change is to gen fall through block first to make sure predecessor is interpreted before successor. This only helps in thunk archetype as each thunk archetype (except those of the leaf MethodHandle) contains customization logic which looks like the following, where the fall through is predecessor of the branch target.
```
                if (ILGenMacros.isShareableThunk()) {
                        undoCustomizationLogic(next);
                        undoCustomizationLogic(filters);
                }   
                if (!ILGenMacros.isCustomThunk()) {
                        doCustomizationLogic();
                }   
```

However, this won't help when we have more complicated control flow, such as nested if statement. So a separate item will be created to do a reverse post order traversal at CFG level.

Reply:
Okay that sounds important since it can reduce the effectiveness of the interpretation that the whole scheme depends on.
Can you please open that issue and make sure it is on the plan to be worked on by talking to Daryl ?

Reply:
Will do